presto-main/src/main/java/com/facebook/presto/operator/WindowFunctionDefinition.java

```diff
@@ -30,27 +30,19 @@
 
     public static WindowFunctionDefinition window(WindowFunctionSupplier functionSupplier, Type type, List<Integer> inputs)
     {
-        requireNonNull(functionSupplier, "functionSupplier is null");
-        requireNonNull(type, "type is null");
-        requireNonNull(inputs, "inputs is null");
-
         return new WindowFunctionDefinition(functionSupplier, type, inputs);
     }
 
     public static WindowFunctionDefinition window(WindowFunctionSupplier functionSupplier, Type type, Integer... inputs)
     {
-        requireNonNull(functionSupplier, "functionSupplier is null");
-        requireNonNull(type, "type is null");
-        requireNonNull(inputs, "inputs is null");
-
         return window(functionSupplier, type, Arrays.asList(inputs));
     }
 
     WindowFunctionDefinition(WindowFunctionSupplier functionSupplier, Type type, List<Integer> argumentChannels)
     {
-        this.functionSupplier = functionSupplier;
-        this.type = type;
-        this.argumentChannels = argumentChannels;
+        this.functionSupplier = requireNonNull(functionSupplier, "functionSupplier is null");
+        this.type = requireNonNull(type, "type is null");
+        this.argumentChannels = requireNonNull(argumentChannels, "inputs is null");
```

Comment:
There was no defensive copy here before and this commit was only about removing redundant `requireNonNulls` but sure, let's make this class even better place. :P
Added in 838717a23c38ccdbc87b7c1009bf488ad31b814a.